### PR TITLE
fix(920274): false positive with CloudFlare header CF-Visitor

### PR DESCRIPTION
**Pull request by dburianov**
REQUEST_HEADERS:CF-Visitor contain { \"scheme\":\"https\"} and requests blocked by this rule

### CODE REVIEW

**Comment by airween** [comment]
Hi @dburianov,

thanks for your fix.

Could you modify the title of the PR and add the affected rule number? Or put the rule id into the comment at least?

Also would be fine to add a new (negative) test to the [collection](https://github.com/coreruleset/coreruleset/blob/v4.0/dev/tests/regression/tests/REQUEST-920-PROTOCOL-ENFORCEMENT/920274.yaml) against this modification.

Thanks!

**Review by airween — APPROVED** [review]
LGTM

**Comment by theseion** [comment]
@dburianov if you want, you can add your name to the list of test contributors at the top of the test file you've modified in this PR.

File: tests/regression/tests/REQUEST-920-PROTOCOL-ENFORCEMENT/920274.yaml [hunk 1]
```diff
@@ -92,3 +92,17 @@ tests:
               Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
           output:
             no_log_contains: "id \"920274\""
+  - test_title: 920274-7
```
Comment (theseion) [inline tests/regression/tests/REQUEST-920-PROTOCOL-ENFORCEMENT/920274.yaml:95]:
```suggestion
  - test_title: 920274-7
    desc: Ignore CloudFlare's "CF-Visitor" header
```

**Review by theseion — CHANGES_REQUESTED** [review]

**Comment by RedXanadu** [comment]
I have some concerns about this change… which I will write up shortly when I have 5 spare minutes… :cold_sweat:

**Comment by RedXanadu** [comment]
For additional context, here's an example false positive, using the example header value from [Cloudflare's documentation](https://developers.cloudflare.com/fundamentals/reference/http-request-headers/#cf-visitor):

``
[Tue Oct 10 12:00:58.033437 2023] [security2:error] [pid 65:tid 140187321493184] [client 172.18.0.1:38712] [client 172.18.0.1] ModSecurity: Warning. Found 6 byte(s) in REQUEST_HEADERS:CF-Visitor outside range: 32,34,38,42-59,61,65-90,95,97-122. [file "/etc/modsecurity.d/owasp-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "1783"] [id "920274"] [msg "Invalid character in request headers (outside of very strict set)"] [data "REQUEST_HEADERS:CF-Visitor={ \\x5c\\x22scheme\\x5c\\x22:\\x5c\\x22https\\x5c\\x22}"] [severity "CRITICAL"] [ver "OWASP_CRS/4.0.0-rc1"] [tag "modsecurity"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [tag "paranoia-level/4"] [hostname "localhost"] [uri "/"] [unique_id "ZSU8-vesKcopyW_rXnUlIgAAAII"]
``

``
[Tue Oct 10 12:00:58.037119 2023] [security2:error] [pid 65:tid 140187321493184] [client 172.18.0.1:38712] [client 172.18.0.1] ModSecurity: Warning. Unconditional match in SecAction. [file "/etc/modsecurity.d/owasp-crs/rules/RESPONSE-980-CORRELATION.conf"] [line "96"] [id "980170"] [msg "Anomaly Scores: (Inbound Scores: blocking=5, detection=5, per_pl=0-0-0-5, threshold=5) - (Outbound Scores: blocking=0, detection=0, per_pl=0-0-0-0, threshold=4) - (SQLI=0, XSS=0, RFI=0, LFI=0, RCE=0, PHPI=0, HTTP=0, SESS=0, COMBINED_SCORE=5)"] [ver "OWASP_CRS/4.0.0-rc1"] [tag "modsecurity"] [tag "reporting"] [hostname "localhost"] [uri "/"] [unique_id "ZSU8-vesKcopyW_rXnUlIgAAAII"]
``

Rule 920274 doesn't like the curly brackets ``{}`` or the backslashes.

**Review by RedXanadu — COMMENTED** [review]
(Reviewed. Concerns as noted above.)

**Comment by airween** [comment]
>     * `CF-Visitor` is not a standard header defined in any RFC or W3C standard (unlike `Referer`, `Sec-CH-UA`, and others). This is not a change to fix a false positive with generic HTTP traffic.

...

Based on @RedXanadu's arguments I have to revise my previous opinion. I think he is right, and we do not want to merge this modification.

> One final thought: a Cloudflare rule exclusion plugin, or a blog post on "how to tune CRS for Cloudflare at higher paranoia levels" or similar 🤔

Yes, I thought about the new CF exclusion plugin instead of this modification.

**Comment by theMiddleBlue** [comment]
agree, this is not a good idea

**Comment by theseion** [comment]
I also agree. Thanks @RedXanadu, great catch! But I'm curious now, @dburianov, are you really running CRS at PL4?

**Comment by theseion** [comment]
Closing this issue due to inactivity. There doesn't appear to be an immediate need to create an exclusion plugin for Cloudflare either.

**Pull request closed without merging**